enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,430 @@
+
+```
+# HELP cluster_installation_create represents a single run of create cluster by the OpenShift installer.
+# TYPE cluster_installation_create histogram
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="15"} 0
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="20"} 0
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="25"} 0
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="30"} 1
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="35"} 1
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="40"} 1
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="45"} 1
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="50"} 1
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="55"} 1
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="60"} 1
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="+Inf"} 1
+cluster_installation_create_sum{platform="aws",result="success",version="4.2",os="linux"} 30
+cluster_installation_create_count{platform="aws",result="success",version="4.2",os="linux"} 1
+```
+The example above represents a single sample. This sample indicates that a Linux user successfully installed a 4.2 AWS cluster, which took between 30 and 35 minutes to complete installation. 
+
+###### Label Values
+* `result` - provides limited context to the outcome of an attempted install:
+    - `ProvisioningFailed`
+    - `BootstrapFailed`
+    - `APIFailed`
+    - `OperatorsDegraded`
+    - `Success`
+* `platform` - currently we deploy to 8 platforms: `aws`, `azure`, `gcp,` `libvirt`, `none`, `openstack`, `ovirt`, and `vsphere`. The list will grow. 
+* `os` - You can find all the available combinations of [Go OS and arch values](https://github.com/golang/go/blob/master/src/go/build/syslist.go) with `go tool dist list`. At this stage I assume we only want to colllect operating systems and some of the values are unlikely/impossible:
+  - `linux`
+  - `darwin` - this could be rewritten to `macos` on the installer client if we prefer
+  - `bsd` - this would encompass the `freebsd`, `netbsd`, and `openbsd` values
+  - `windows` - future proofing? 
+  - `other` - everything else not captured above
+* `version` - represents minor releases; assuming we do not backport metrics, `4.5` will be the only value if we make it into the next release and we could expect roughly 20 values over the next five years.
```

Comment:
Existing metrics like `cluster_version` include the full version string in their `version` label (e.g. [here][1]).  Why would we not do the same here?

Also, where are we getting this version information from?  Currently `cluster_version` is using the version information baked into the `release-metadata` file in the release image's layers.  But also currently, the release image is an opaque pullspec as far as the installer is concerned.  Should this string be the installer version (leaning on openshift/installer#2682 and assuming no release image overrides)?  Or the pullspec's tag/digest?  Or...?

[1]: https://github.com/openshift/cluster-version-operator/blob/732ae625e991bf1447370b537caf9bcc22c850dc/pkg/cvo/metrics.go#L170

Reply:
>  Why would we not do the same here?

In order to reduce cardinality, see e.g. https://github.com/openshift/enhancements/pull/205/files#diff-3b226c8b21f37f235bcd51fbc7128a3cR268

 > Also, where are we getting this version information from? 
> Should this string be the installer version (leaning on openshift/installer#2682 and assuming no release image overrides)? 

Great question. `version` should definitely be installer version. Will look into  openshilft/installer/#2682 as  a source. thanks for the suggestion.